src/nvim/ex_getln.c

```diff
@@ -2767,11 +2791,13 @@ char *getcmdline(int firstc, int count, int indent, bool do_concat FUNC_ATTR_UNU
 /// @param[in]  xp_context  Type of expansion.
 /// @param[in]  xp_arg  User-defined expansion argument.
 /// @param[in]  highlight_callback  Callback used for highlighting user input.
+/// @param[in]  one_key  Return after one key press for button prompt.
+/// @param[in]  mouse_used  Set to true when returning after right mouse click.
 ///
 /// @return [allocated] Command line or NULL.
 char *getcmdline_prompt(const int firstc, const char *const prompt, const int hl_id,
```

Comment:
@bfredl `cmdline_show` events do not receive this prompt hl_id, so ext_cmdline cannot highlight the prompt accordingly. Is adding this hl_id as an additional parameter warranted?

Reply:
Added in #31661.